### ISSUE TITLE
[BIFT] restrict the q-range for speed and less noisy data

### ISSUE THREAD

**Issue by kif** [kif/freesas#18]
Mark suggests to start at the begining of the guinier region: reasonable

But where to stop ? dq*Dmax<2pi ?

**Comment by kif** [comment]
Mark wants the full noise: great :)

**Comment by maaeli** [comment]
Hi Jérôme,

I am not surprised as the "Dmax" information mostly comes for the small angles.
Do you have test data with low-q artifacts? Because there dropping low q might be the game changer.  Looking here https://www.sasbdb.org/browse having no artifact seems to be the exception...

Cheers,
Martha

**Comment by kif** [comment]
Hi Martha,

Nice to read you. How is your new life?

I am also disappointed by those poor results. Here are a couple of images of the same BSA dataset, reconstructed from the (B)IFT transformed data whether the region before Guinier's linerar region was discarded or not. Of course, it is always possible that there was an error in my code.
![image](https://user-images.githubusercontent.com/1018880/84500800-8ba31400-acb5-11ea-8408-2aca509247ba.png)
![image](https://user-images.githubusercontent.com/1018880/84500822-9493e580-acb5-11ea-8a9c-fd21157818be.png)

It is not really surprising when considering the Dmax is completely wrong after BIFT transformation:
![image](https://user-images.githubusercontent.com/1018880/84500928-db81db00-acb5-11ea-87cd-db2e36645f20.png)
![image](https://user-images.githubusercontent.com/1018880/84500938-e0468f00-acb5-11ea-97df-80b3f52303e1.png)

**Comment by maaeli** [comment]
Hi Jérôme, 

how do you obtain those plots? Because for me, the bift.py do look like superimposed by a zig-zag...

[bsa_005_sub.out.zip](https://github.com/kif/freesas/files/4774433/bsa_005_sub.out.zip)

Anyways, the second data above looks a lot like smeared out sphere? 

Best,
Martha

**Comment by kif** [comment]
I agree with you, the second solution looks like a sense sphere. This can be understood as the algorithm starts with a sphere and an alpha_max parameter which would make this acceptable. 
Then starts the optimization process. 
Different scenarii have been investigated. The one currently implemented is:
* One log scan on alpha (between 1/alpha_max and alpha_max) with Dmax=3Rg
* One linear scan on Dmax from 2 to 4 Rg
Of course the Guinier region has better to be acceptable from beginning :(
The 2D grid scan has been dropped as it was too time consuming.
Then a steepest descent (Powell) to find the best alpha/Dmax.
The monte-carlo is at the end, it is just to get some statistics around the best solution.

Are you using the provided test-data ? because I got this:
![image](https://user-images.githubusercontent.com/1018880/84747818-133f9a00-afb8-11ea-920e-068c7979fadd.png)
(this is the result of `freesas testdata/bsa_005_sub.dat`)

**Comment by maaeli** [comment]
Yes, I am using the provided test data...
Btw, how do you get the reciprocal space fits?

**Comment by kif** [comment]
IFT is an ill-posed inverse problem. This means the opposite is trivial, trivial like a dot product :)
I wrote plenty of test based on synthetic data, have a look at the test_autorg there is a test_synthetic.
https://github.com/kif/freesas/blob/fit_autorg/freesas/test/test_autorg.py#L68
```
        R0 = 4
        npt = 1000
        I0 = 1e2
        Dmax = 2 * R0
        size = 5000
        r = numpy.linspace(0, Dmax, npt + 1)
        p = distribution_sphere(I0, Dmax, npt)
        q = numpy.linspace(0, 10, size)
        qr = numpy.outer(q, r / pi)
        T = (4 * pi * (r[-1] - r[0]) / npt) * numpy.sinc(qr)
        I = T.dot(p)
        err = numpy.sqrt(I)
        data = numpy.vstack((q, I, err)).T
```

**Comment by maaeli** [comment]
Hi Jérôme, 

for me BIFT  works on the BSA test set, even if I crop the pre-Guinier data:
![image](https://user-images.githubusercontent.com/4751755/86509650-cb0edd00-bde9-11ea-84cc-e359d02c5aec.png)
Fit for complete data:
![image](https://user-images.githubusercontent.com/4751755/86509664-db26bc80-bde9-11ea-953b-0b8a8f2b57c0.png)
Fit for cropped data:
![image](https://user-images.githubusercontent.com/4751755/86509668-e4178e00-bde9-11ea-8cad-842d5359be84.png)

How exactly did you remove the pre-Guinier region?

**Comment by kif** [comment]
You are right, it looks OK... I don't know what I did implement
